src/org/zaproxy/zap/extension/ascanrulesBeta/InsecureHTTPMethod.java

```diff
@@ -1,455 +1,702 @@
+        TreeSet<HtmlParameter> cookies = tracemsg.getCookieParams();
+        cookies.add(new HtmlParameter(HtmlParameter.Type.cookie, randomcookiename, randomcookievalue));
+        tracemsg.setCookieParams(cookies);
+
+        sendAndReceive(tracemsg, false); // do not follow redirects. That might
+        // ruin our day.
+        // if the response *body* from the TRACE request contains the cookie,
+        // we're in business :)
+        if (tracemsg.getResponseBody().toString().contains(randomcookievalue)) {
+            return new MessageAndEvidence(tracemsg, randomcookievalue);
+        }
+
+        return null;
+    }
+
+    private boolean testConnect(HttpMessage baseMsg, String thirdpartyHost, int thirdpartyPort, Pattern thirdPartyContentPattern) throws Exception {
+
+        String connecthost = baseMsg.getRequestHeader().getURI().getHost();
+        int connectport = baseMsg.getRequestHeader().getURI().getPort();
+
+        // this cannot currently be done using the existing HttpSender class, so
+        // do it natively using HttpClient,
+        // in as simple as possible a manner.
+        Socket socket = null;
+        try {
+            ProxyClient client = new ProxyClient();
+            client.getHostConfiguration().setProxy(connecthost, connectport);
+            client.getHostConfiguration().setHost(thirdpartyHost, thirdpartyPort);
+            ConnectResponse connectResponse = client.connect();
+            StatusLine statusLine = connectResponse.getConnectMethod().getStatusLine();
+
+            if (log.isDebugEnabled())
+                log.debug("The status line returned: " + statusLine);
+
+            int statusCode = statusLine.getStatusCode();
+            socket = connectResponse.getSocket();
+            if (socket != null && statusCode == HttpStatus.SC_OK) {
+                // we have a socket and a 200 status.
+                // Could still be a false positive though, if the server ignored
+                // the method,
+                // and did not recognise the URL, so redirected to a login page,
+                // for instance
+                // Remediation: Check the contents match the expected third
+                // party contents.
+                if (log.isDebugEnabled())
+                    log.debug("Raw Socket established, in theory to " + thirdpartyHost);
+
+                OutputStream os = socket.getOutputStream();
+                InputStream is = socket.getInputStream();
+
+                PrintWriter pw = new PrintWriter(os, false);
+                pw.write("GET http://" + thirdpartyHost + ":" + thirdpartyPort + "/ HTTP/1.1\n");
+                pw.write("Host: " + thirdpartyHost + "\n\n");
+                pw.flush();
+
+                // read the response via a 4k buffer
+                ByteArrayOutputStream bos = new ByteArrayOutputStream();
+                byte[] buffer = new byte[1024 * 4];

```

Comment:
Is there a reason to do this math on the fly? I know this is totally insignificant performance wise but really......4096

Reply:
I have no idea why it was written like that, I thought he wanted to clarify that it was 4k but its pretty obvious.
I didn't modify what was already working but I should have.
I'll fix it now :)